src/main/java/org/broadinstitute/consent/http/service/dao/DatasetServiceDAO.java

```diff
@@ -365,15 +366,25 @@ public void executeUpdateDatasetWithFiles(Handle handle,
       List<DatasetProperty> properties,
       List<FileStorageObject> uploadedFiles,
       boolean executeDeletes) {
-    addAuditRecord(datasetId, datasetName, userId, AuditActions.UPDATE);
-    // update dataset
-    datasetDAO.updateDatasetByDatasetId(
-        datasetId,
-        datasetName,
-        new Timestamp(new Date().getTime()),
-        userId,
-        dacId
-    );
+
+    // Don't update the name if it isn't provided
+    if (StringUtils.isBlank(datasetName)) {
+      Dataset dataset = datasetDAO.findDatasetById(datasetId);
```

Comment:
Instead of a different code path here, what about
```suggestion
    if (StringUtils.isBlank(datasetName)) {
      datasetName = datasetDAO.findDatasetById(datasetId).datasetName;
    }
    addAuditRecord(datasetId, datasetName, userId, AuditActions.UPDATE);
    // update dataset
    datasetDAO.updateDatasetByDatasetId(
    ...
```